pyhub/mcptools/google/sheets/auth.py

```
@@ -0,0 +1,171 @@
+                success_message="인증 성공! 이 창은 닫으셔도 됩니다.",
+                open_browser=True,
+            )
+
+            # 토큰 저장
+            self._save_credentials(credentials)
+            logger.info("인증이 완료되었습니다.")
+
+            return credentials
+
+        except Exception as e:
+            raise AuthenticationError(f"인증 중 오류 발생: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)